services/semaphore.js

```diff
@@ -7,20 +8,41 @@ module.exports = {
 		return Boolean(env.SEMAPHORE);
 	},
 	configuration({env, cwd}) {
-		const pr = env.PULL_REQUEST_NUMBER;
-		const isPr = Boolean(pr);
+		if (env.BRANCH_NAME) {
+			// Exists in v1 only
+			const pr = env.PULL_REQUEST_NUMBER;
+			const isPr = Boolean(pr);
+			return {
+				name: 'Semaphore',
+				service: 'semaphore',
+				commit: head({env, cwd}),
+				build: env.SEMAPHORE_BUILD_NUMBER,
+				branch: isPr ? undefined : env.BRANCH_NAME,
+				pr,
+				isPr,
+				prBranch: isPr ? env.BRANCH_NAME : undefined,
+				slug: env.SEMAPHORE_REPO_SLUG,
+				root: env.SEMAPHORE_PROJECT_DIR,
+			};
+		}
 
+		let parsedRepo;
+
+		try {
+			parsedRepo = parseGitUrl(env.SEMAPHORE_GIT_URL || '');
+		} catch (error) {
+			console.log(error);
+		}
+
+		// V2
 		return {
 			name: 'Semaphore',
 			service: 'semaphore',
-			commit: head({env, cwd}),
-			build: env.SEMAPHORE_BUILD_NUMBER,
-			branch: isPr ? undefined : env.BRANCH_NAME,
-			pr,
-			isPr,
-			prBranch: isPr ? env.BRANCH_NAME : undefined,
-			slug: env.SEMAPHORE_REPO_SLUG,
-			root: env.SEMAPHORE_PROJECT_DIR,
+			build: env.SEMAPHORE_WORKFLOW_ID,
+			branch: env.SEMAPHORE_GIT_BRANCH,
```

Comment:
`branch` is the Git branch being built or targeted by a Pull Request.
Have you verified that `SEMAPHORE_GIT_BRANCH` is set to the branch being build for non-PR and the target for PR builds?

Reply:
I ran tests in both PRs and non-PRs in V2. `SEMAPHORE_GIT_BRANCH` is:
- the branch being built for non-PR
- the branch being built for PRs
There is no idea of target branch, unfortunately. We cannot even identify what is a PR and what is not, so it would not be possible to decide to set branch OR prBranch. What behaviour do you want to adopt?

Reply:
Then `SEMAPHORE_GIT_BRANCH` should be used only for `branch`. Basically all builds on V2 will be considered non PR builds and `isPR`/`prBranch` will not be supported for V2.
That should be added to the doc.